medicines/import/src/report.rs

```diff
@@ -0,0 +1,105 @@
+struct Uploaded {
+    file_name: String,
+    hash: String,
+}
+
+struct SkippedDuplicate {
+    file_name: String,
+    hash: String,
+}
+
+struct SkippedIncomplete {
+    file_path: String,
+}
+
+struct SkippedUnreleased {
+    file_name: String,
+    release_state: String,
+}
+
+pub(crate) struct Report {
+    uploaded: Vec<Uploaded>,
+    skipped_duplicates: Vec<SkippedDuplicate>,
+    skipped_incompletes: Vec<SkippedIncomplete>,
+    skipped_unreleaseds: Vec<SkippedUnreleased>,
+}
+
+impl Report {
+    pub(crate) fn new() -> Report {
+        Report {
+            uploaded: Vec::new(),
+            skipped_duplicates: Vec::new(),
+            skipped_incompletes: Vec::new(),
+            skipped_unreleaseds: Vec::new()
+        }
+    }
+
+    pub(crate) fn add_uploaded(&mut self, file_name: &str, hash: &str) {
+        self.uploaded.push(Uploaded {
+            file_name: file_name.to_string(),
+            hash: hash.to_string()
+        });
+    }
+
+    pub(crate) fn add_skipped_duplicate(&mut self, file_name: &str, hash: &str) {
+        self.skipped_duplicates.push(SkippedDuplicate {
+            file_name: file_name.to_string(),
+            hash: hash.to_string()
+        });
+    }
+
+    pub(crate) fn add_skipped_incomplete(&mut self, file_path: &str) {
+        self.skipped_incompletes.push(SkippedIncomplete {
+            file_path: file_path.to_string()
+        });
+    }
+
+    pub(crate) fn add_skipped_unreleased(&mut self, file_name: &str, release_state: &str) {
+        self.skipped_unreleaseds.push(SkippedUnreleased {
+            file_name: file_name.to_string(),
+            release_state: release_state.to_string()
+        });
+    }
+
+    pub(crate) fn already_uploaded_file_with_hash(&self, hash: &str) -> bool {
+        self.uploaded.iter().any(|uploaded| uploaded.hash == hash)
```

Comment:
This is cool! And fine as we're not that performance sensitive, but was thinking that if we were, we could use a `HashMap` and check and add in one go:
```rust
use std::collections::HashMap;
let mut map = HashMap::new();
map.entry(hash).or_insert(true);
```

Reply:
Would probably want to match on the `Entry` so we can add to the report if `Entry::Occupied`. Anyway, not really important

Reply:
Yeah, if we were dealing with a dataset of like 5 million I would definitely have gone for the hashmap here.

Reply:
cool